tasks/gulp-transifex-test.js

```diff
@@ -0,0 +1,182 @@
+import fs      from 'fs';
+import _       from 'lodash';
+import nconf   from 'nconf';
+import gulp    from 'gulp';
+import request from 'superagent';
+
+nconf.argv().env().file({ file: 'config.json' });
+
+const LOCALES = './common/locales/';
+const ENGLISH_LOCALE = `${LOCALES}en/`;
+const ALL_LANGUAGES = getArrayOfLanguages();
+
+const SLACK_URL = nconf.get('TRANSIFEX_SLACK:url');
+const SLACK_CHANNEL = '#' + (nconf.get('TRANSIFEX_SLACK:channel') || 'general');
+const SLACK_USERNAME = 'Transifex';
+const SLACK_EMOJI = ':transifex:';
+
+gulp.task('transifex', ['transifex:missingFiles', 'transifex:missingStrings', 'transifex:malformedStrings']);
+
+gulp.task('transifex:missingFiles', () => {
+
+  let missingStrings = [];
+
+  eachTranslationFile(ALL_LANGUAGES, (error) => {
+    if(error) {
+      missingStrings.push(error.path);
+    }
+  });
+
+  if (!_.isEmpty(missingStrings)) {
+    let message = 'the following files were missing from the translations folder';
+    post(message, missingStrings);
+  }
+});
+
+gulp.task('transifex:missingStrings', () => {
+
+  let missingStrings = [];
+
+  eachTranslationString(ALL_LANGUAGES, (language, filename, key, englishString, translationString) => {
+    if (!translationString) {
+      let errorString = `${language} - ${filename} - ${key} - ${englishString}`;
+      missingStrings.push(errorString);
+    }
+  });
+
+  if (!_.isEmpty(missingStrings)) {
+    let message = 'The following strings are not translated';
+    post(message, missingStrings);
+  }
+});
+
+gulp.task('transifex:malformedStrings', () => {
+
+  let jsonFiles = stripOutNonJsonFiles(fs.readdirSync(ENGLISH_LOCALE));
+  let interpolationRegex = /<%= [a-zA-Z]* %>/g;
+  let stringsToLookFor = getStringsWith(jsonFiles, interpolationRegex);
+
+  let stringsWithMalformedInterpolations = [];
+  let stringsWithIncorrectNumberOfInterpolations = [];
+
+  let count = 0;
+  _(ALL_LANGUAGES).each(function(lang) {
+
+    _.each(stringsToLookFor, function(strings, file) {
+      let translationFile = fs.readFileSync(LOCALES + lang + '/' + file);
+      let parsedTranslationFile = JSON.parse(translationFile);
+
+      _.each(strings, function(value, key) {
+        let translationString = parsedTranslationFile[key];
+        if (!translationString) return;
+
+        let englishOccurences = stringsToLookFor[file][key];
+        let translationOccurences = translationString.match(interpolationRegex);
+
+        if (!translationOccurences) {
+          let malformedString = `${lang} - ${file} - ${key} - ${translationString}`;
+          stringsWithMalformedInterpolations.push(malformedString);
+        } else if (englishOccurences.length !== translationOccurences.length) {
+          let missingInterploationString = `${lang} - ${file} - ${key} - ${translationString}`;
+          stringsWithIncorrectNumberOfInterpolations.push(missingInterploationString);
+        }
+      });
+    });
+  });
+
+  if (!_.isEmpty(stringsWithMalformedInterpolations)) {
```

Comment:
Should this isEmpty check be nested inside the post function? Or is it easier to understand outside of it?